instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts

```diff
@@ -8,6 +8,8 @@ group = "io.opentelemetry.instrumentation"
 val versions: Map<String, String> by project
 
 dependencies {
+  compileOnly(project(":instrumentation-annotation-support"))
```

Comment:
Hmm, should we use `implementation` here? It probably should be built in the autoconfigure module by default, we probably shouldn't force users to add this dependency by themselves.

Reply:
Maybe `api` instead of `implementation`?

Reply:
Hm, `api` would work too, but do we expect anybody to use those classes directly? If not then `implementation` would suffice